### PR TITLE
Updated error returns on rmw_take

### PR DESCRIPTION
**Pull request by Lobotuerk**
Signed-off-by: lobotuerk <jtlorente@ekumenlabs.com>

### CODE REVIEW

**Comment by Lobotuerk** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12300/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7272/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10016/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12198/)

**Comment by Lobotuerk** [comment]
Aarch64 does not allow Connext

File: rmw_connext_cpp/src/rmw_take.cpp [hunk 1]
```diff
@@ -148,23 +148,22 @@ _take(
   DDS::InstanceHandle_t * sending_publication_handle,
   rmw_subscription_allocation_t * allocation)
 {
-  if (!subscription) {
-    RMW_SET_ERROR_MSG("subscription handle is null");
-    return RMW_RET_ERROR;
-  }
+  RMW_CHECK_FOR_NULL_WITH_MSG(
+    subscription, "subscription info is null",
```
Comment (Blast545) [inline rmw_connext_cpp/src/rmw_take.cpp:152]: Why info instead of handle/pointer?
```suggestion
    subscription, "subscription handle is null",
```

**Review by Blast545 — COMMENTED** [review]
Minor suggestion + you'll have to disable the test for aarch64 to have green CI in the 4 platforms before merging. Otherwise LGTM

**Review by hidmic — APPROVED** [review]
LGTM with @Blast545 comments addressed.

**Comment by Lobotuerk** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12362/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7331/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10074/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12274/)

File: rmw_connext_cpp/src/rmw_take.cpp [hunk 2]
```diff
@@ -215,38 +214,37 @@ _take_sequence(
   rmw_subscription_allocation_t * allocation)
 {
   (void) allocation;
-  if (!subscription) {
-    RMW_SET_ERROR_MSG("subscription handle is null");
-    return RMW_RET_INVALID_ARGUMENT;
-  }
+
+  RMW_CHECK_ARGUMENT_FOR_NULL(
+    subscription, RMW_RET_INVALID_ARGUMENT);
```
Comment (Blast545) [inline rmw_connext_cpp/src/rmw_take.cpp:219]: Why this one doesn't return a msg and the one for `rmw_take` does? Same questions for the rest of the checks in `take_seqeuence`

**Review by Blast545 — APPROVED** [review]
Minor question, LGTM

File: rmw_connext_cpp/src/rmw_take.cpp (hunk 2 above)
Comment (Lobotuerk) [reply rmw_connext_cpp/src/rmw_take.cpp:219]: Great question. Actually, I forgot to get rid of the messages on the first ones. I got rid of all messages as @hidmic suggested, they were kind of random depending on each implementation, and pretty inaccurate, the default `RMW_CHECK_ARGUMENT_FOR_NULL` seems more appropriate.

**Pull request merged**